lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -103,6 +106,19 @@ class DiscoverLoops {
             recordDataflow(ref, data);
             recordProbe(data, ref);
           })
+          .Case<RWProbeOp>([&](RWProbeOp rwProbe) {
+            hw::InnerSymTarget refTarget = irn.lookup(rwProbe.getTarget());
+            if (!refTarget) {
+              rwProbe->emitWarning("cannot resolve target innerRef");
+              return;
+            }
+            FieldRef ref = getRefForIST(refTarget);
+            if (!ref)
+              return;
+            auto data = rwProbe.getResult();
+            addDrivenBy({data, 0}, ref);
```

Comment:
Does this work if the rwprobe target isn't a leaf (multiple FieldRef's)?

(not sure how this works, but do we create edge for all fields connected?)

Reply:
Consider:
```
FIRRTL version 3.3.0
circuit Bar :
  module Foo :
    input agg : { x: Clock, y : Clock }
    output ref : RWProbe<{ x: Clock, y : Clock }>
    output refx : RWProbe<Clock>

    define ref = rwprobe(agg)
    define refx = rwprobe(agg.x)

  module Bar :
    input clock : Clock

    inst foo of Foo
    connect foo.agg.x, clock
    connect foo.agg.y, clock

    force(clock, UInt<1>(1), foo.ref.x, read(foo.refx))
```

Which should report a cycle but doesn't.